low-level/message_handlers/node_data_msg_handler.py

```diff
@@ -720,20 +714,21 @@ def _generate_cpu_data(self):
                     # Store the state to Persistent Cache.
                     self.persist_state_data('cpu', 'CPU_USAGE_DATA')
 
-    def _send_ifdata_json_msg(self, sensor_type, resource_id, resource_type, state, severity, event=""):
+    def _send_ifdata_json_msg(self, resource_id, resource_type, state, event=""):
         """A resuable method for transmitting IFDataMsg to RMQ and IEM logging"""
-        ifDataMsg = IFdataMsg(self._node_sensor.host_id,
-                                self._node_sensor.local_time,
-                                self._node_sensor.if_data,
-                                resource_id,
-                                resource_type,
-                                state, severity, event)
+        if_data = [inf for inf in self._node_sensor.if_data \

```

Comment:
So till now all nw interfaces data were getting packed in, instead of the one for which alert is generated ? This is better approach then